pallets/registry/src/benchmarking.rs

```diff
@@ -62,15 +62,189 @@ benchmarks! {
 		);
 		let registry_id = Ss58Identifier::to_registry_id(&(id_digest).encode()[..]).unwrap();
 
-		Pallet::<T>::create(origin.clone(), registry, None).expect("Should create a registry entry");
+		//Pallet::<T>::create(origin.clone(), registry, None).expect("Should create a registry entry");
+        Pallet::<T>::create(origin.clone(), registry, Option::None).expect("Should create a registry entry");
 
 		let registry_update: Vec<u8> = (2u8..u8::MAX).cycle().take(T::MaxEncodedRegistryLength::get().try_into().unwrap()).collect();
 		let utx_registry: InputRegistryOf::<T> = BoundedVec::try_from(registry_update)
 			.expect("Update Registry should fit into the expected input length of the test runtime");
+
+
 	}: _<T::RuntimeOrigin>(origin, utx_registry, registry_id.clone())
 	verify {
 		assert_last_event::<T>(Event::Update { registry: registry_id, authority: did }.into());
 	}
+ 
+archive {
+        let l in 1 .. MAX_REGISTRY_SIZE;
+
+		let caller: T::AccountId = account("caller", 0, SEED);
+		let did: T::RegistryCreatorId = account("did", 0, SEED);
+
+		let origin =  <T as Config>::EnsureOrigin::generate_origin(caller.clone(), did.clone());
+
+		let raw_registry: Vec<u8> = (0u8..u8::MAX).cycle().take(T::MaxEncodedRegistryLength::get().try_into().unwrap()).collect();
+		let registry: InputRegistryOf::<T> = BoundedVec::try_from(raw_registry)
+			.expect("Test Registry should fit into the expected input length of the test runtime");
+		let digest = <T as frame_system::Config>::Hashing::hash(&registry[..]);
+		let id_digest = <T as frame_system::Config>::Hashing::hash(
+			&[&registry.encode()[..], &did.encode()[..]].concat()[..],
+		);
+		let registry_id = Ss58Identifier::to_registry_id(&(id_digest).encode()[..]).unwrap();
+
+		Pallet::<T>::create(origin.clone(), registry, None).expect("Should create a registry entry");
+
+		let registry_archive: Vec<u8> = (2u8..u8::MAX).cycle().take(T::MaxEncodedRegistryLength::get().try_into().unwrap()).collect();
+		let utx_registry: InputRegistryOf::<T> = BoundedVec::try_from(registry_archive)
```

Comment:
registry_ archive and utx_registry variables are not required here

Reply:
Done